src/service/config.py

```diff
@@ -0,0 +1,55 @@
+"""
+Configuration settings for the CDM MCP Server.
+
+This service enables AI assistants to interact with Delta Lake tables stored in MinIO through Spark,
+implementing the Model Context Protocol (MCP) for natural language data operations.
+"""
+
+import logging
+import os
+from functools import lru_cache
+
+from pydantic import BaseModel, Field
+
+APP_VERSION = "0.1.0"
+
+
+class Settings(BaseModel):
+    """
+    Application settings for the CDM MCP Server.
+    """
+
+    app_name: str = "CDM MCP Server"
+    app_description: str = (
+        "FastAPI service for AI assistants to interact with Delta Lake tables via Spark"
+    )
+    api_version: str = APP_VERSION
+    log_level: str = Field(
+        default=os.getenv("LOG_LEVEL", "INFO"),
+        description="Logging level for the application",
+    )
+
+
+@lru_cache()
+def get_settings() -> Settings:
+    """
+    Get the application settings.
+
+    Uses lru_cache to avoid loading the settings for every request.
+    """
+    return Settings()
+
+
+def configure_logging():
+    """Configure logging for the application."""
+    settings = get_settings()
+    if settings.log_level.upper() not in logging.getLevelNamesMapping():
+        logging.warning(
+            "Unrecognized log level '%s'. Falling back to 'INFO'.",
+            settings.log_level,
```

Comment:
I'm not sure if this will priint if basicconfig() hasn't been run yet

Reply:
I think it would still print - just not the format set via basicconfig()

Reply:
From what I recall working on the CTS, it didn't print at all before the config was run. I might be misremembering though